### PR TITLE
fix(cloudformation-diff): move aws-sdk to dependency for cfn-diff to get CFN types resolved in exports

### LINKED ISSUE #28679: cloudformation-diff: Hidden dependency on aws-sdk
### Describe the bug

We have a custom aws-cdk consruct library that exports cloudformation-diff. 

Since `@aws-cdk/cloudformation-diff` for version `2.119.0`, it is throwing the following error:
```
node_modules/@aws-cdk/cloudformation-diff/lib/diff-template.d.ts:1:32 - error TS2307: Cannot find module 'aws-sdk' or its corresponding type declarations.

1 import { CloudFormation } from 'aws-sdk';
                                 ~~~~~~~~~


Found 1 error in node_modules/@aws-cdk/cloudformation-diff/lib/diff-template.d.ts:1
```

We believe the error is coming from changes in this PR: https://github.com/aws/aws-cdk/pull/28336

Specifically in `packages/@aws-cdk/cloudformation-diff/lib/diff-template.ts`, the following change: 
```
// The SDK is only used to reference `DescribeChangeSetOutput`, so the SDK is added as a devDependency.
// The SDK should not make network calls here
// eslint-disable-next-line import/no-extraneous-dependencies
import { CloudFormation } from 'aws-sdk';
```

I have a sample repo to reproduce the exact issue: https://github.com/frankpengau/cdk-lib-cfn-diff-issue-20240112

### Expected Behavior

It should not have a missing dependency. 

### Current Behavior

It currently has an indirect dependency on `aws-sdk` causing our construct library to have a transitive dependency on `aws-sdk`. If it is indeed a dependency, it should be added in as such, instead of being only a devdependency. 

### Reproduction Steps

1. `cdk init lib --language typescript`
2. Install `@aws-cdk/cloudformation-diff` for version `2.119.0`
3. In a new file in lib folder: `export * as cfndiff from @aws-cdk/cloudformation-diff`
4. Compile typescript `tsc`

### Possible Solution

Explicit dependency on `aws-sdk`, so that it shows up in the `package-lock.json` file and installs it as necessary for `@aws-cdk/cloudformation-diff` to work. 

### Additional Information/Context

_No response_

### CDK CLI Version

2.114.1

### Framework Version

_No response_

### Node.js Version

18.13.0

### OS

macOS

### Language

TypeScript

### Language Version

TypeScript (5.2.2)

### Other information

_No response_

### PR DESCRIPTION
**Pull request by frankpengau**
Issue still persists after #28680. 
```
node_modules/@aws-cdk/cloudformation-diff/lib/diff-template.d.ts:1:37 - error TS2307: Cannot find module 'aws-sdk' or its corresponding type declarations.

1 import type { CloudFormation } from 'aws-sdk';
                                      ~~~~~~~~~


Found 1 error in node_modules/@aws-cdk/cloudformation-diff/lib/diff-template.d.ts:1
```

Types are still required as a direct dependency in package.json.

Closes #28679 

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by frankpengau** [comment]
Exemption Request

**Comment by frankpengau** [comment]
@pahud Csn I trouble you to get someone to review this? The previous PR linked in the description did not resolve the issue.

**Comment by frankpengau** [comment]
@kylelaker @comcalvi Any other suggestions? Otherwise, based off other usage in the repo for import type, going to try and push ahead for this change.

**Review by paulhcsun — DISMISSED** [review]
Hi @frankpengau, looks like the codebuild is failing and is requiring you to just run `yarn install` to install a missing dependency.

**Comment by frankpengau** [comment]
@paulhcsun Thanks for looking at this PR! 

Currently on extended AL for another week and don't have my laptop with me. 

If urgent, happy for you to take over.

If not, happy to address things once I'm back in a week's time. :)

**Comment by paulhcsun** [comment]
Hi @frankpengau, I've pushed the yarn.lock changes. Could you explain why the `aws-sdk` version was downgraded to `"2.1548.0"`?

**Comment by frankpengau** [comment]
Sorry @paulhcsun , just got around to this, updated yarn lock and also version to latest on main. Please re-review at your earliest convenience! Many Thanks

**Comment by frankpengau** [comment]
I think these pipeline errors are coming from upstream changes? Probably this merged PR: https://github.com/aws/aws-cdk/pull/29471 with similar [build log error](https://2cl995f30a.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=915d4509-2698-43a8-a828-9f41698fc207%2Fbuild.log)?

File: packages/@aws-cdk/cloudformation-diff/package.json [hunk 1]
```diff
@@ -25,6 +25,7 @@
   "dependencies": {
     "@aws-cdk/aws-service-spec": "^0.0.53",
     "@aws-cdk/service-spec-types": "^0.0.53",
+    "aws-sdk": "2.1562.0",
```
Comment (paulhcsun) [inline packages/@aws-cdk/cloudformation-diff/package.json:28]:
```suggestion
    "aws-sdk": "2.1576.0",
```

**Review by paulhcsun — APPROVED** [review]

**Pull request merged**